ivy/array/creation.py

```diff
@@ -1,8 +1,92 @@
 # global
 import abc
+from typing import Optional, Union, List
 
-# ToDo: implement all methods here as public instance methods
+# local
+import ivy
+
+# Array API Standard #
+# -------------------#
 
 
 class ArrayWithCreation(abc.ABC):
-    pass
+    def asarray(
+        self: ivy.Array,
+        *,
+        copy: Optional[bool] = None,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+    ) -> ivy.Array:
+        return ivy.asarray(self._data, copy=copy, dtype=dtype, device=device)
+
+    def full_like(
+        self: ivy.Array,
+        fill_value: Union[int, float],
+        *,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        return ivy.full_like(
+            self._data, fill_value, dtype=dtype, device=device, out=out
+        )
+
+    def ones_like(
+        self: ivy.Array,
+        *,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        return ivy.ones_like(self._data, dtype=dtype, device=device, out=out)
+
+    def zeros_like(
+        self: ivy.Array,
+        *,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        return ivy.zeros_like(self._data, dtype=dtype, device=device, out=out)
+
+    def tril(self: ivy.Array, k: int = 0, out: Optional[ivy.Array] = None) -> ivy.Array:
+        return ivy.tril(self._data, k, out=out)
+
+    def triu(self: ivy.Array, k: int = 0, out: Optional[ivy.Array] = None) -> ivy.Array:
+        return ivy.triu(self._data, k, out=out)
+
+    def empty_like(
+        self: ivy.Array,
+        *,
+        dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
+        device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        return ivy.empty_like(self._data, dtype=dtype, device=device, out=out)
+
+    def meshgrid(
+        self: ivy.Array,
+        *arrays: Union[ivy.Array, ivy.NativeArray],
+        indexing: Optional[str] = "xy",
+    ) -> List[ivy.Array]:
+        list = [self._data]
+        for array in arrays:
+            list.append(array._data)
```

Comment:
this is wrong, the arrays might be `ivy.NativeArray` instances in which case this will fail. You can just do `list = [self._data] + list(arrays)`.

Reply:
Should we be calling ._data on the arrays argument like with self?